code/modules/antagonists/voidwalker/voidwalker_mob.dm

```diff
@@ -0,0 +1,336 @@
+	var/rclick_damage_type = BRUTE
+	/// Can we speak?
+	var/can_speak = FALSE
+	/// Turf that our abilities rely on. VVing this automatically sets all the components and removes space dependency stuff if the new tile isnt space
+	var/home_turf = /turf/open/space
+	/// Decal that we can kidnap on
+	var/kidnapping_decal = /obj/effect/decal/cleanable/vomit/nebula
+	/// Toggle for abduction interactions, in-case we have non-kidnap subtypes ()
+	var/can_do_abductions = TRUE
+
+/mob/living/basic/voidwalker/Initialize(mapload, mob/tamer)
+	ADD_TRAIT(src, TRAIT_FREE_HYPERSPACE_MOVEMENT, INNATE_TRAIT) //Need to set before init cause if we init in hyperspace we get dragged before the trait can be added
+	. = ..()
+
+	AddElement(/datum/element/simple_flying)
+	AddElement(/datum/element/glass_pacifist)
+
+	AddComponent(/datum/component/regenerator, brute_per_second = 2, burn_per_second = 2, outline_colour = regenerate_colour, regen_check = CALLBACK(src, PROC_REF(can_regen)))
+	AddComponent(/datum/component/glass_passer, deform_glass = 5 SECONDS)
+	AddComponent(/datum/component/planet_allergy)
+
+	AddElement(/datum/element/dextrous, hud_type = hud_type)
+	AddComponent(/datum/component/basic_inhands, x_offset = -2)
+
+	AddElement(/datum/element/pick_and_drop_only)
+
+	AddElement(/datum/element/glass_bleeder)
+
+	charge = new charge(src)
+	charge.Grant(src)
+
+	// Glass passing is handled by the glass passer component
+	passtable_on(src, type)
+
+	// Voidwalker lore is that radio's actually attracted them, so they should be able to listen to it
+	var/obj/item/radio/internal_radio = new /obj/item/radio(src)
+	internal_radio.keyslot = /obj/item/encryptionkey/heads/captain
+	internal_radio.subspace_transmission = TRUE
+	internal_radio.canhear_range = 0 // anything higher and people in the area will hear it too
+	internal_radio.recalculateChannels()
+
+	telepathy =  new telepathy(src)
+	telepathy.Grant(src)
+
+	unique_setup()
+
+/// Stuff you might want different on subtypes
+/mob/living/basic/voidwalker/proc/unique_setup()
+	AddComponent(/datum/component/space_camo, space_alpha, non_space_alpha, 255, 5 SECONDS, image(icon, icon_state + "_stealthed", ABOVE_LIGHTING_PLANE))
+	AddComponent(/datum/component/space_dive, /obj/effect/dummy/phased_mob/space_dive/voidwalker)
+
+	unsettle = new(src)
+	unsettle.Grant(src)
+
+	fully_replace_character_name(null, pick(GLOB.voidwalker_names))
+
+/mob/living/basic/voidwalker/Destroy()
+	QDEL_NULL(unsettle)
+	QDEL_NULL(telepathy)
+	QDEL_NULL(charge)
+
+	return ..()
+
+/// Called on COMSIG_LIVING_UNARMED_ATTACK
+/mob/living/basic/voidwalker/early_melee_attack(atom/target, list/modifiers, ignore_cooldown)
+	. = ..()
+
+	if(!. || !can_do_abductions)
+		return
+
+	if(ishuman(target))
+		var/mob/living/carbon/human/hewmon = target
+
+		var/should_attack = try_kidnap(hewmon)
+
+		if(!should_attack)
+			return FALSE
+
+		if(hewmon.stat == HARD_CRIT && !hewmon.has_trauma_type(/datum/brain_trauma/voided))
```

Comment:
Only just thought about this because people were talking about Memento Mori exploits but what about the edge cases like zombies where a human continues walking around and acting while in crit? Not sure what the trait was called but you should be able to check for it, alternately you can just leave it as is and it's probably fine.